### DISCUSSION TITLE
Crazyflie firmware with windows

### DISCUSSION THREAD

**Discussion by noammmmmm** [bitcraze/discussions#86] [Q&A]
Hello

I am trying to update the firmware of the crazyflie with Visual Studio Code.
I am using Windows 10 and wsl.
I don't want to use the Virtual Machine.
The libusb is installed with Zadig.
The python works fine and the cfclient works fine with the Dongle.
When i am typing the command "make cload", I am receiving the message:

noam@DESKTOP-JPACDKC:/mnt/c/NoamFolder/crazyflie/crazyflie-firmware$ make cload
tools/kbuild/Makefile.kbuild:147: warning: overriding recipe for target 'cload'
Makefile:140: warning: ignoring old recipe for target 'cload'
make[1]: Entering directory '/mnt/c/NoamFolder/crazyflie/crazyflie-firmware/build'
python3 -m cfloader  flash  cf2.bin stm32-fw
Restart the Crazyflie you want to bootload in the next
 10 seconds ...
Failed to flash: Cannot find a Crazyradio Dongle
make[1]: Leaving directory '/mnt/c/NoamFolder/crazyflie/crazyflie-firmware/build'

What can be the proble?

**Comment by knmcguire** [comment]
Hi! Thanks for providing the information!

I would have expected to see the following, as it supposed to use the 'windows python' to flash the firmware, like this.
`python.exe -m cfloader  flash  cf2.bin stm32-fw
`

First some questions first to debug your issue:

- Is your system configured for WSL2 ?
- Do you see the same thing if you have the firmware folder locally in WSL? So in your /home/ folder and not /mnt/c/...

**Comment by noammmmmm** [comment]
Hi

Thank you
The
python.exe -m cfloader flash cf2.bin stm32-fw
works fine
Actually, i would prefer not to work with WSL/ubuntu at all.
I didn't try to work with /home/ folder but
python.exe -m cfloader flash cf2.bin stm32-fw will satisfy me for now.
Now, I would like to work with the debugger.
I bought the debugging adaptor from your site.
I am working with Visual Studio Code
I want to work only with windows and not with VM.
I followed the instructions in:
https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/openocd_gdb_debugging/

And I received the message:
[cid:5d28b3cf-8616-496f-88e7-b301b7c0d407]

My launch.json file is as described in the link.
My working directory is:
***@***.***:/mnt/c/NoamFolder/crazyflie/crazyflie-firmware$
Should i move the files:

openocd
arm-none-eabi-gcc

To the working directory?

Where does it save them after i downloaded them by:

sudo apt-get install openocd
sudo apt-get install gcc-arm-none-eabi gdb-multiarch
sudo ln -s /usr/bin/gdb-multiarch /usr/local/bin/arm-none-eabi-gdb

Thanks

Noam
On-chip debugging | Bitcraze<https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/openocd_gdb_debugging/>
“svdFile” refers to the necessary file for peripheral registers to show up nicely in the debug pane, all named and structured; we’ll add it in the next step
www.bitcraze.io





________________________________

**Reply by knmcguire** [reply]
I understand with not working with the VM as that one can be slow, but anything for compiling C and debugging is difficult to setup on Windows... we used to advise Cygwin to people but that one had too many issues. And now ever since WSL exists... that is just too convenient for windows development that we  rather want people to use that. And with the remote connection of vs-code it's as simple as coding natively (and much faster than the VM)

Connecting an USB is not standard in WSL, but I use myself this tool called USBipd, which allows me to either connect a crazyradio or a programmer directly in WSL itself: https://github.com/dorssel/usbipd-win/wiki/WSL-support. I highly recommend you to use that, and then follow the GDB debugging instructions after that.

If you are still committed to do development on windows directly for the firmware, then I'm afraid that we need some time to investigate how to do that.

**Comment by noammmmmm** [comment]
Thank you for the answer

Now I am trying to use VM again. It worked in the past.
But after a while i get the message:

Not in a hypervisor partition (HVP=0) (VERR_NEM_NOT_AVAILABLE).

VT-x is disabled in the BIOS for all CPU modes (VERR_VMX_MSR_ALL_VMX_DISABLED).

And the VM did not open.
Do you think I need to update the VM?

Thanks again

Noam

________________________________

**Comment by noammmmmm** [comment]
Another issue

When the CF is ok
Power on and all is good: The blue LEDs (2 and 3) are fully lit and the front right LED (1) is blinking red twice every second.

But i get:
Power on and all is good but sensors are not yet calibrated: The blue LEDs (2 and 3) are fully lit and the front right LED (1) is blinking red with 2 seconds interval. Put the Crazyflie 2.X on a level surface and keep it absolutely still to calibrate.

What should I do to calibrate it?

Thanks
Noam
________________________________

**Comment by knmcguire** [comment]
@noammmmmm the two questions above are not really related to your original question. To keep it maintainable for, could you please make 2 new discussions, each containing your new questions above
